pkg/steps/multi_stage.go

```diff
@@ -231,6 +251,27 @@ func addSecret(secret string, pod *coreapi.Pod) {
 	})
 }
 
+func addProfile(name string, profile api.ClusterProfile, pod *coreapi.Pod) {
+	volumeName := "cluster-profile"
+	pod.Spec.Volumes = append(pod.Spec.Volumes, coreapi.Volume{
+		Name: volumeName,
+		VolumeSource: coreapi.VolumeSource{
+			Secret: &coreapi.SecretVolumeSource{
+				SecretName: name,
+			},
+		},
+	})
+	container := &pod.Spec.Containers[0]
+	container.VolumeMounts = append(container.VolumeMounts, coreapi.VolumeMount{
+		Name:      volumeName,
+		MountPath: clusterProfileMountPath,
+	})
+	container.Env = append(container.Env, coreapi.EnvVar{
+		Name:  "CLUSTER_TYPE",
+		Value: strings.Split(string(profile), "-")[0],
```

Comment:
Ouch. Is this (rather arbitrary) dependency of cluster type on a pattern of the cluster profile name documented somewhere? I'm almost tempted to ask if we should have a `(*api.ClusterProfile).toClusterType()` for this to avoid burying this dependency here.

Reply:
I don't know if we'll gain much in terms of overall maintenance, but I like the idea of having that in `api`. Added.

Reply:
I imagined some poor soul creating a `aws_awesome` cluster profile and the following debugging session resulting in discovering this line :)